engine.js

```diff
@@ -72,23 +94,26 @@ var handleUniqueLoanNumberErrors = function(counts) {
     return errors;
 };
 
-var readResponseSync = function(APIURL, funcName, year, params) {
+var apiGET = function(APIURL, funcName, year, params) {
     var url = APIURL + '/' + funcName + '/' + year;
     for (var i = 0; i < params.length; i++) {
         url = url + '/' + params[i];
     }
```

Comment:
Not part of this review, but you could rewrite this loop as `url =  url + '/' + params.join('/');`. Little cleaner, and maybe a small performance increase if you ever have a lot of params.